dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderResource.java

```diff
@@ -106,4 +114,94 @@ public final Response loadFolderByURI(@Context final HttpServletRequest httpServ
         return response;
     }
 
+    /**
+     * <p>
+     * This endpoint returns a folder structure with their children recursively
+     * </p>
+     * @param siteName (site or host) and folder (name of the folder)
+     * @return a folder structure with their children recursively
+     * @see <a href="https://github.com/dotCMS/core/issues/18964">Please check the github issue!</a>
+     */
+    @GET
+    @Path ("/sitename/{siteName}/folder/{folder : .+}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response loadFolderChildrenByURIPath(@Context final HttpServletRequest httpServletRequest,
+                                                      @Context final HttpServletResponse httpServletResponse,
+                                                      @PathParam("siteName") final String siteName,
+                                                      @PathParam("folder") final String uri) throws DotDataException, DotSecurityException, DotDataException, DotSecurityException   {
+        Response response = null;
+        final InitDataObject initData = this.webResource.init(null, httpServletRequest, httpServletResponse, true, null);
+        final User user = initData.getUser();
+        final String uriParam = !uri.startsWith(StringPool.FORWARD_SLASH) ? StringPool.FORWARD_SLASH.concat(uri) : uri;
+        final Host host = APILocator.getHostAPI().findByName(siteName, user, false);
+        //final Folder folder = folderHelper.loadFolderByURI(siteName,user,uriParam);
+        final Folder folder = APILocator.getFolderAPI().findFolderByPath(uriParam, host, user, false);
+        if(host==null || folder==null) {
+            throw new DoesNotExistException("No folder found for "+uri+" on site "+siteName);
+        }
+        CustomFolderView root = getFolderStructure(folder, user);
+        response = Response.ok( new ResponseEntityView(root) ).build();
+        return response;
+    }
+
+    /**
+     * <p>
+     * This method returns a folder structure with their children recursively based on
+     * the folder returned by findFolderByPath
+     * </p>
+     * @param Folder (folder from findFolderByPath) and User (logged in user)
+     * @return CustomFolderView a folder structure with their children recursively
+     */
+    private final CustomFolderView getFolderStructure(Folder folder, User user){
+
+        CustomFolderView customFolder = convertFrom(folder);
+
+        List<CustomFolderView> foldersChildCustoms = new LinkedList<>();
+        List<Folder> children = null;
+        try {
+            children = APILocator.getFolderAPI().findSubFolders(folder, user, false);
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5916479)